src/go/k8s/controllers/redpanda/cluster_controller.go

```diff
@@ -237,6 +237,14 @@ func (r *ClusterReconciler) Reconcile(
 		sts,
 	}
 
+	result, errs := ar.Ensure()
+	if !result.IsZero() {
+		return result, nil
+	}
+	if errs != nil {
+		return result, errs
+	}
```

Comment:
This seems not intuitive. Why errs is checked after result not before?

Cane result be not zero and error not nil?

Reply:
Yes, result can be `ctrl.Result{RequeueAfter: e.RequeueAfter}` without an error and it take precedence over an error result as it will have a specific RequeueAfter instead of a error backoff.